## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 410672`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/15232258365


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8852au</li>
    <li>linuxPackages_5_4_hardened.rtl8852au</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_5_10_hardened.rtl8852au (linuxKernel.packages.linux_5_10_hardened.rtl8852au)</li>
    <li>linuxPackages_5_15_hardened.rtl8852au (linuxKernel.packages.linux_5_15_hardened.rtl8852au)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.rtl8852au</li>
    <li>linuxKernel.packages.linux_5_15.rtl8852au</li>
    <li>linuxKernel.packages.linux_5_4.rtl8852au</li>
    <li>linuxKernel.packages.linux_6_1.rtl8852au</li>
    <li>linuxPackages.rtl8852au (linuxKernel.packages.linux_6_12.rtl8852au)</li>
    <li>linuxPackages_hardened.rtl8852au (linuxPackages_6_12_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_13.rtl8852au</li>
    <li>linuxPackages_6_13_hardened.rtl8852au (linuxKernel.packages.linux_6_13_hardened.rtl8852au)</li>
    <li>linuxPackages_latest.rtl8852au (linuxKernel.packages.linux_6_14.rtl8852au)</li>
    <li>linuxPackages_6_14_hardened.rtl8852au (linuxKernel.packages.linux_6_14_hardened.rtl8852au)</li>
    <li>linuxPackages_6_1_hardened.rtl8852au (linuxKernel.packages.linux_6_1_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_6.rtl8852au</li>
    <li>linuxPackages_6_6_hardened.rtl8852au (linuxKernel.packages.linux_6_6_hardened.rtl8852au)</li>
    <li>linuxPackages_ham.rtl8852au (linuxKernel.packages.linux_ham.rtl8852au)</li>
    <li>linuxPackages_latest-libre.rtl8852au (linuxKernel.packages.linux_latest_libre.rtl8852au)</li>
    <li>linuxPackages-libre.rtl8852au (linuxKernel.packages.linux_libre.rtl8852au)</li>
    <li>linuxPackages_lqx.rtl8852au (linuxKernel.packages.linux_lqx.rtl8852au)</li>
    <li>linuxPackages_xanmod.rtl8852au (linuxKernel.packages.linux_xanmod.rtl8852au)</li>
    <li>linuxPackages_xanmod_latest.rtl8852au (linuxKernel.packages.linux_xanmod_latest.rtl8852au, linuxPackages_xanmod_stable.rtl8852au)</li>
    <li>linuxPackages_zen.rtl8852au (linuxKernel.packages.linux_zen.rtl8852au)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxPackages_5_10_hardened.rtl8852au</summary>
<pre>  343 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_btc.c:343:44: error: invalid initializer
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/scripts/Makefile.build:286: /build/source/phl/phl_cmd_btc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
/build/source/phl/phl_cmd_ps.c: In function 'phl_register_ps_module':
/build/source/phl/phl_cmd_ps.c:1504:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
 1504 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_ps.c:1504:44: error: invalid initializer
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/scripts/Makefile.build:286: /build/source/phl/phl_cmd_ps.o] Error 1
make[2]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/Makefile:1840: /build/source] Error 2
make[1]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/Makefile:192: __sub-make] Error 2
make: *** [Makefile:639: modules] Error 2</pre>
</details>
<details>
<summary>linuxPackages_5_15_hardened.rtl8852au</summary>
<pre>/build/source/phl/phl_cmd_btc.c:343:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
  343 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_btc.c:343:44: error: invalid initializer
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/scripts/Makefile.build:289: /build/source/phl/phl_cmd_btc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/build/source/phl/phl_cmd_ps.c: In function 'phl_register_ps_module':
/build/source/phl/phl_cmd_ps.c:1504:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
 1504 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_ps.c:1504:44: error: invalid initializer
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/scripts/Makefile.build:289: /build/source/phl/phl_cmd_ps.o] Error 1
make[1]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/Makefile:1914: /build/source] Error 2
make: *** [Makefile:639: modules] Error 2</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 12 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8852au</li>
    <li>linuxKernel.packages.linux_lqx.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod_stable.rtl8852au</li>
    <li>linuxKernel.packages.linux_zen.rtl8852au</li>
    <li>linuxPackages_5_4_hardened.rtl8852au</li>
    <li>linuxPackages_lqx.rtl8852au</li>
    <li>linuxPackages_xanmod.rtl8852au</li>
    <li>linuxPackages_xanmod_latest.rtl8852au</li>
    <li>linuxPackages_xanmod_stable.rtl8852au</li>
    <li>linuxPackages_zen.rtl8852au</li>
  </ul>
</details>
<details>
  <summary>:x: 18 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.rtl8852au</li>
    <li>linuxPackages_5_10_hardened.rtl8852au (linuxKernel.packages.linux_5_10_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_5_15.rtl8852au</li>
    <li>linuxPackages_5_15_hardened.rtl8852au (linuxKernel.packages.linux_5_15_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8852au</li>
    <li>linuxKernel.packages.linux_6_1.rtl8852au</li>
    <li>linuxPackages.rtl8852au (linuxKernel.packages.linux_6_12.rtl8852au)</li>
    <li>linuxPackages_hardened.rtl8852au (linuxPackages_6_12_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_13.rtl8852au</li>
    <li>linuxPackages_6_13_hardened.rtl8852au (linuxKernel.packages.linux_6_13_hardened.rtl8852au)</li>
    <li>linuxPackages_latest.rtl8852au (linuxKernel.packages.linux_6_14.rtl8852au)</li>
    <li>linuxPackages_6_14_hardened.rtl8852au (linuxKernel.packages.linux_6_14_hardened.rtl8852au)</li>
    <li>linuxPackages_6_1_hardened.rtl8852au (linuxKernel.packages.linux_6_1_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_6.rtl8852au</li>
    <li>linuxPackages_6_6_hardened.rtl8852au (linuxKernel.packages.linux_6_6_hardened.rtl8852au)</li>
    <li>linuxPackages_ham.rtl8852au (linuxKernel.packages.linux_ham.rtl8852au)</li>
    <li>linuxPackages_latest-libre.rtl8852au (linuxKernel.packages.linux_latest_libre.rtl8852au)</li>
    <li>linuxPackages-libre.rtl8852au (linuxKernel.packages.linux_libre.rtl8852au)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_10.rtl8852au</summary>
<pre>Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/15n0gdj62y8qz7gnwgsgx1yqx71hl1m3-115.patch
patching file os_dep/linux/ioctl_cfg80211.c
patching file os_dep/linux/ioctl_cfg80211.c
patching file os_dep/osdep_service_linux.c
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/s1qkj0ph0ma64a6743mvkwnabrbw1hsc-bash-5.2p37/bin/bash ARCH=arm64 CONFIG_PLATFORM_I386_PC=n CONFIG_PLATFORM_ARM_RPI=y
#rm -f .symvers.8852au
make ARCH=arm64 CROSS_COMPILE= -C  M=/build/source  modules
make[1]: Entering directory '/build/source'
make[1]: *** M=/build/source: No such file or directory.  Stop.
make[1]: Leaving directory '/build/source'
make: *** [Makefile:639: modules] Error 2</pre>
</details>
</details>

--- a/pkgs/os-specific/linux/rtl8852au/default.nix
+++ b/pkgs/os-specific/linux/rtl8852au/default.nix
@@ -60,13 +60,18 @@ stdenv.mkDerivation (finalAttrs: {
     nuke-refs $out/lib/modules/*/kernel/net/wireless/*.ko
   '';
 
+  # GCC 14 makes this an error by default
+  env.NIX_CFLAGS_COMPILE = "-Wno-designated-init";
+
   enableParallelBuilding = true;
 
   meta = with lib; {
     description = "Driver for Realtek 802.11ac, rtl8852au, provides the 8852au mod";
     homepage = "https://github.com/lwfinger/rtl8852au";
     license = licenses.gpl2Only;
-    platforms = platforms.linux;
+    platforms = [ "x86_64-linux" ];
+    # FIX: error: invalid initializer
+    broken = kernel.kernelOlder "6" && kernel.isHardened;
     maintainers = with maintainers; [ lonyelon ];
   };
 })

--- a/pkgs/os-specific/linux/rtl8852au/default.nix
+++ b/pkgs/os-specific/linux/rtl8852au/default.nix
@@ -27,6 +27,11 @@ stdenv.mkDerivation (finalAttrs: {
     "format"
   ];
 
+  patches = [
+    # https://github.com/lwfinger/rtl8852au/pull/115
+    ./fix-build-for-kernels-6.13-6.14.patch
+  ];
+
   postPatch = ''
     substituteInPlace ./Makefile \
       --replace-fail /sbin/depmod \# \

--- a/pkgs/os-specific/linux/rtl8852au/fix-build-for-kernels-6.13-6.14.patch
+++ b/pkgs/os-specific/linux/rtl8852au/fix-build-for-kernels-6.13-6.14.patch
@@ -1,0 +1,77 @@
+From c65ed43f42656aecf43e7ea80c58d204c3c67aca Mon Sep 17 00:00:00 2001
+From: Soham Nandy <soham.nandy2006@gmail.com>
+Date: Fri, 28 Mar 2025 17:24:55 +0530
+Subject: [PATCH 1/2] rtl8852au(fix): remove MODULE_IMPORT and net_device for
+ kernel versions over 6.13
+
+---
+ os_dep/linux/ioctl_cfg80211.c | 3 +++
+ os_dep/osdep_service_linux.c  | 4 ++++
+ 2 files changed, 7 insertions(+)
+
+diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
+index 2b79c97..277dffb 100755
+--- a/os_dep/linux/ioctl_cfg80211.c
++++ b/os_dep/linux/ioctl_cfg80211.c
+@@ -6350,6 +6350,9 @@ static void rtw_get_chbwoff_from_cfg80211_chan_def(
+ 
+ static int cfg80211_rtw_set_monitor_channel(struct wiphy *wiphy
+ #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0))
++	, struct net_device *dev
++#endif
+ 	, struct cfg80211_chan_def *chandef
+ #else
+ 	, struct ieee80211_channel *chan
+diff --git a/os_dep/osdep_service_linux.c b/os_dep/osdep_service_linux.c
+index fe47c3b..8fdbcfc 100644
+--- a/os_dep/osdep_service_linux.c
++++ b/os_dep/osdep_service_linux.c
+@@ -390,7 +390,9 @@ static int openFile(struct file **fpp, const char *path, int flag, int mode)
+ 	struct file *fp;
+ 
+ #if defined(MODULE_IMPORT_NS)
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 13, 0))
+ 	MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
++#endif
+ #endif
+ 
+ 	fp = filp_open(path, flag, mode);
+@@ -508,7 +510,9 @@ static int isFileReadable(const char *path, u32 *sz)
+ 	char buf;
+ 
+ #if defined(MODULE_IMPORT_NS)
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 13, 0))
+ 	MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
++#endif
+ #endif
+ 
+ 	fp = filp_open(path, O_RDONLY, 0);
+
+From 91d168fc5aa818b4e85aa5b2b43d7f25470e925c Mon Sep 17 00:00:00 2001
+From: Soham Nandy <soham.nandy2006@gmail.com>
+Date: Mon, 7 Apr 2025 10:25:03 +0530
+Subject: [PATCH 2/2] rtl8852au(fix): get_tx_power callback by adding link_id
+ parameter
+
+kernel versions >6.14 cfg80211_ops was updated to include an unsigned
+int link_id parameter.
+---
+ os_dep/linux/ioctl_cfg80211.c | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
+index 277dffb..3d7620e 100755
+--- a/os_dep/linux/ioctl_cfg80211.c
++++ b/os_dep/linux/ioctl_cfg80211.c
+@@ -4454,6 +4454,10 @@ static int cfg80211_rtw_set_txpower(struct wiphy *wiphy,
+ static int cfg80211_rtw_get_txpower(struct wiphy *wiphy,
+ #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
+ 	struct wireless_dev *wdev,
++	
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,14,0))
++	unsigned int link_id,
++#endif
+ #endif
+ 	int *dbm)
+ {

